syntax = "proto3";

package medical_qa;

option go_package = "llm-qa-system/proto";

// Main QA service definition
service MedicalQAService {
    // Generate a draft answer for medical questions
    rpc GenerateDraftAnswer (QuestionRequest) returns (QuestionResponse) {}
    // Review and modify draft answer
    rpc ReviewAnswer (ReviewRequest) returns (ReviewResponse) {}
}

message QuestionRequest {
    string question_id = 1;
    string question_text = 2;
    string patient_context = 3;
    string medical_category = 4;
}

message QuestionResponse {
    string question_id = 1;
    string draft_answer = 2;
    repeated string references = 3;
    float confidence_score = 4;
}

message ReviewRequest {
    string question_id = 1;
    string draft_answer = 2;
    string reviewer_id = 3;
    string reviewer_comments = 4;
    bool is_approved = 5;
}

message ReviewResponse {
    string question_id = 1;
    string final_answer = 2;
    string review_status = 3;
    string reviewer_id = 4;
    string review_timestamp = 5;
}